src/coreclr/vm/threadsuspend.cpp

```diff
@@ -358,12 +355,6 @@ Thread::SuspendThreadResult Thread::SuspendThread(BOOL fOneTryOnly, DWORD *pdwSu
                 STRESS_LOG1(LF_SYNC, LL_INFO1000, "In Thread::SuspendThread ::SuspendThread returned %x\n", dwSuspendCount);
             }
 
-            // Our callers generally expect that STR_Failure means that
-            // the thread has exited.
-#ifndef TARGET_UNIX
-            _ASSERTE(NtCurrentTeb()->LastStatusValue != STATUS_SUSPEND_COUNT_EXCEEDED);
```

Comment:
`LastStatusValue` is not a documented field. It is better to avoid depending on it in any way, even it is just assert.